language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -1 +1,1875 @@
-include!(concat!(env!("OUT_DIR"), "/syntax.rs"));
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use codespan::{ByteIndex, Span};
+use std::fmt;
+use std::str::FromStr;
+
+use crate::ast::{
+    parse_field, BinOp, Block, Block_, Builtin, Cmd, CopyableVal, CopyableVal_, Exp, Exp_, Field_,
+    Fields, Function, FunctionBody, FunctionCall, FunctionCall_, FunctionName, FunctionVisibility,
+    Function_, IfElse, ImportDefinition, Kind, LValue, LValue_, Loop, ModuleDefinition,
+    ModuleIdent, ModuleName, Program, QualifiedModuleIdent, QualifiedStructIdent, Script,
+    ScriptOrModule, Spanned, Statement, StructDefinition, StructDefinition_, StructName, Type,
+    TypeVar, TypeVar_, UnaryOp, Var, Var_, While,
+};
+use crate::lexer::*;
+use hex;
+use libra_types::{account_address::AccountAddress, byte_array::ByteArray};
+
+// FIXME: The following simplified version of ParseError copied from
+// lalrpop-util should be replaced.
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ParseError<L, T, E> {
+    InvalidToken {
+        location: L,
+    },
+    UnrecognizedToken {
+        token: (L, T, L),
+        expected: Vec<String>,
+    },
+    User {
+        error: E,
+    },
+}
+
+/// Format a list of expected tokens.
+fn fmt_expected(f: &mut fmt::Formatter, expected: &[String]) -> fmt::Result {
+    if !expected.is_empty() {
+        writeln!(f)?;
+        for (i, e) in expected.iter().enumerate() {
+            let sep = match i {
+                0 => "Expected one of",
+                _ if i < expected.len() - 1 => ",",
+                // Last expected message to be written
+                _ => " or",
+            };
+            write!(f, "{} {}", sep, e)?;
+        }
+    }
+    Ok(())
+}
+
+impl<L, T, E> fmt::Display for ParseError<L, T, E>
+where
+    L: fmt::Display,
+    T: fmt::Display,
+    E: fmt::Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::ParseError::*;
+        match *self {
+            User { ref error } => write!(f, "{}", error),
+            InvalidToken { ref location } => write!(f, "Invalid token at {}", location),
+            UnrecognizedToken {
+                token: (ref start, ref token, ref end),
+                ref expected,
+            } => {
```

Comment:
you don't need the `ref` on each pattern match if you remove the `*` on self

Reply:
This is part of the code that I copied over from lalrpop. We should redo the error handling and replace this code.